apps/arena/lib/arena/game/player.ex

```diff
@@ -124,14 +132,34 @@ defmodule Arena.Game.Player do
     put_in(player, [:aditional_info, :stamina_interval], stamina_interval)
   end
 
+  def recover_mana(player) do
+    now = System.monotonic_time(:millisecond)
+    time_since_last = now - player.aditional_info.mana_recovery_time_last_at
+
+    if time_since_last >= player.aditional_info.mana_recovery_time_interval_ms do
```

Comment:
Won't this value be nil in case we choose a damage recovery strategy? but the game is not crashing. Can you explain this pls?

Reply:
Got lucky on this one (a.k.a not planned), I should make an explicit check on the type of mana recovery (time or damage based)

As to why it works, well its due to [term ordering](https://hexdocs.pm/elixir/Kernel.html#module-term-ordering), in essence any 2 terms in Elixir can be compared and there is a specific ordering if compering different terms
```
number < atom < reference < function < port < pid < tuple < map < list < bitstring
```

In this particular case we are comparing a number (`now`) and an atom (`nil`), so it will always return false